### PR TITLE
More AQA fixes

### PR DESCRIPTION
**Pull request by SchwartzGandhi**
apparently Miepee doesn't read these nice changelog descriptions I write out for him so screw me I guess

### CODE REVIEW

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 1]
```diff
@@ -222,8 +222,7 @@ Extra - room_id: [6]
           Can Use Power Bombs
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: For the sake of generator health, I would rather make this connection completely trivial, and put all the requirements that are needed to leave the item, on the way to the item.
So that door -> pb tank:
```md
ALL:
- can use any bomb
- ANY:
  - Grav and Damage run (bg) and Electric damage >= 20
  - Damage run (int) and Electric damage >= 80
```

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 2]
```diff
@@ -257,8 +256,13 @@ Extra - room_id: [6]
   * Layers: default
   > Door to Sciser Stall
       Any of the following:
-          # Destroy bomb blocks by standing off of crab: https://youtu.be/h8Q6TKHg8kM
-          Hi-Jump and Stand On Frozen Enemies (Beginner) and Can Freeze Enemies and Can Use Bombs and Can Use Springball
+          All of the following:
+              # Destroy bomb blocks by standing off of crab: https://youtu.be/h8Q6TKHg8kM
+              Stand On Frozen Enemies (Beginner) and Can Freeze Enemies and Can Use Bombs
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:261]: This needs spring ball, or mid air morph in order to break the blocks.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 3]
```diff
@@ -257,8 +256,13 @@ Extra - room_id: [6]
...
+              Any of the following:
+                  Hi-Jump
+                  # NHJ: https://youtu.be/_F80ASGzrvo
+                  Movement (Advanced) and Wall Jump (Beginner)
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:265]: I would just make it movement beginner and use the ledge grab video: https://youtu.be/6RhuTs1mD1c

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 4]
```diff
@@ -278,7 +282,7 @@ Extra - room_id: [6]
               # TODO: damage values
               Gravity Suit and Damage Runs (Beginner)
   > Other to Serris Escape
-      Can Climb High Jump Wall
+      Trivial
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:285]: nice. i assume it was this jump?
grafik

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 5]
```diff
@@ -545,9 +549,6 @@ Extra - room_id: [14]
   * Layers: default
   * L0 Hatch to Cheddar Bay/Door to Aquarium Shaft West
   * Extra - door_idx: (33,)
-  > Door to Worst Room in the Game
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:548]: my phrasing in the note was bad.
Secret message jr should stay here. The grav+SJ should move onto this connection.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 6]
```diff
@@ -681,7 +687,7 @@ Extra - room_id: [17]
   * L0 Hatch to Aquarium Shaft West/Door to Aquarium Tank
   * Extra - door_idx: (39,)
   > Pickup (Missile Tank)
-      Can Use Power Bombs
+      Can Kill Tough Beam-Resistant Enemy and Can Use Power Bombs
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:690]: IDK if this requirement should be here. its very easy to just fall down after laying the pb and instantly collect the item without needing to interact with the pirate.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 7]
```diff
@@ -960,8 +967,12 @@ Extra - room_id: [23]
...
   * Extra - door_idx: (53,)
   > Room Center
-      # Secret Message Shinespark
-      Gravity Suit and Hi-Jump and Level 0 Locks and Level 4 Locks and Space Jump and Speed Booster and Shinespark Tricks (Hypermode) and Disabled Door Lock Rando and Disabled Entrance Rando and Can Break Single Bomb Blocks
+      All of the following:
+          # Secret Message Shinespark
+          Gravity Suit and Level 0 Locks and Level 4 Locks and Space Jump and Speed Booster and Shinespark Tricks (Hypermode) and Disabled Door Lock Rando and Disabled Entrance Rando and Can Break Single Bomb Blocks
+          Any of the following:
+              Movement (Expert)
+              Hi-Jump and Movement (Advanced)
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:976]: IMO:
- secret message should be expert shinesparking.
- HJ should not put any additional trick requiremetns
- NHJ should be either hypermode shinesparking, or advanced movement.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 8]
```diff
@@ -1031,7 +1042,8 @@ Extra - room_id: [24]
       All of the following:
           Missiles
           Any of the following:
-              Can Use Any Bombs
+              Can Use Power Bombs
+              Hi-Jump and Can Use Bombs
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1046]: Why is HJ needed? Are you sure you didn't mean the spring ball template instead?

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 9]
```diff
@@ -257,8 +256,13 @@ Extra - room_id: [6]
   * Layers: default
   > Door to Sciser Stall
       Any of the following:
-          # Destroy bomb blocks by standing off of crab: https://youtu.be/h8Q6TKHg8kM
-          Hi-Jump and Stand On Frozen Enemies (Beginner) and Can Freeze Enemies and Can Use Bombs and Can Use Springball
+          All of the following:
+              # Destroy bomb blocks by standing off of crab: https://youtu.be/h8Q6TKHg8kM
+              Stand On Frozen Enemies (Beginner) and Can Freeze Enemies and Can Use Bombs
+              Any of the following:
+                  Hi-Jump
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:263]: This may just be me, I'd rather have you use the template here for future-casing.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 10]
```diff
@@ -1394,16 +1409,25 @@ Extra - room_id: [34]
...
-      Any of the following:
-          Can Use Any Bombs
-          Gravity Suit and Morph Ball and Screw Attack and Knowledge (Beginner)
+      All of the following:
+          Morph Ball
+          Any of the following:
+              Gravity Suit and Screw Attack and Knowledge (Beginner)
+              All of the following:
+                  Can Use Any Bombs
+                  Mid-Air Morph (Beginner) or Can Use Springball
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1430]: grafik
mid air morph is needed to go through the tunnel here because you cant ledge grab it, right?

Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:1430]: correct, hole is only one tile up from the ground

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 5 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:548]: is there any actual point to having it be a certain connection?

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 4 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:285]: correct

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 10 above)
Comment (JeffGainsNGames) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:1430]: bombs + grav also works

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 3 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:265]: imo, using a ledge grab to circumvent a wall jump trick requirement doesn't sound right to me.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 1 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: you want *all* of the requirements to just be on leaving? shouldn't the connection to the pickup at least have a requirement for pbs?

Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: Does the PB block respawn?

Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: no

Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: Also no, you misunderstood. I want *all* the rquirements on getting the item and want *none* on leaving.

Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:222]: oh i see, my mistake

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 3 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:265]: I guess my only other worry is that the timer for the crab to thaw is pretty tight. and imo doing ledge grabs is a lot slower than regular wall jumps

Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:265]: ledge grabs arent really that much slower.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt (hunk 7 above)
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:976]: disagree one first point. Secret message is the longest possible shinespark in the game, and there are so many opportunities for you too mess up, and have to go all the way back to the beginning to start it again. perfectly encapsulates what I hypermode trick should be.
with the other two points tho

Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:976]: Our guideline for hypermode shinesparking is:
> frame perfect charges/conservations, etc

Secret Message is in no way frame perfect. It may have a lot of opportunities to mess it up, but there are also plenty of places where you can still restore the conservation.
It's just nowhere the other hypermode shinesparks that we have, which currently are:
- old bob skip (i think you have like 3 spare frames?)
- Reverse NHJ charge beam etank (iirc you have like 2 spare frames)
- Killing the maintanance shaft geron with a shinespark (very tight *and* you have to do a correlax jump)
- killing the central hub geron NHJ (very tight)


In comparison to all of those, one could almost call Secret Message consistent. It perfectly fits into our general expert guideline too:
> These take a lot of practice, may still fail a few attempts, setting up for next attempt might be slow

TLDR; You can practice Secret Message and get to the point where you can just pick up a controller and do it on less than 3 attempts which IMO does not encapsulate hypermode.

Comment (JeffGainsNGames) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:976]: I would agree with hypermode before Miguel proved the main deck geron spark was possible. To me these 2 tricks definitely could be seperated due to the frame perfect nature of the latter.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 11]
```diff
@@ -1664,15 +1674,23 @@ Extra - room_id: [38]
...
+                      All of the following:
+                          Disabled Entrance Rando
+                          Any of the following:
+                              Speed Booster
+                              # https://youtu.be/ncBn4hyjC-A
+                              Hi-Jump and Wall Jump (Beginner)
+                              # https://youtu.be/AWajsYrYgbA
+                              Stand On Frozen Enemies (Intermediate) and Wall Jump (Intermediate) and Can Freeze Enemies
+              # TODO: vid
+              Hi-Jump and Underwater Wall Jump (Hypermode) and Disabled Entrance Rando
```
[the remark this answers is not in the document]
Comment (SchwartzGandhi) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:1693]: not sure about this, who's to say that the entrance that gets rolled won't be harder than the vanilla?
plus, im pretty sure the fact you go through a transition at all is the reason why its set to hypermode

[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:1693]: Ah. I was only considering cases where the bottom room doesnt have water, but we cant guaranteee that.
Fair enough then.

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 12]
```diff
@@ -1681,13 +1679,13 @@ Extra - room_id: [38]
...
+                      # Wall jump from room below https://youtu.be/ncBn4hyjC-A
+                      Hi-Jump and Wall Jump (Beginner)
                       All of the following:
                           Disabled Entrance Rando
                           Any of the following:
                               Speed Booster
-                              # https://youtu.be/ncBn4hyjC-A
-                              Hi-Jump and Wall Jump (Beginner)
-                              # https://youtu.be/AWajsYrYgbA
+                              # Freeze enemie from room below and wall jump https://youtu.be/AWajsYrYgbA
```
Comment (Miepee) [inline randovania/games/fusion/logic_database/Sector 4 AQA.txt:1688]: *enemies

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 13]
```diff
@@ -434,12 +434,14 @@ Extra - room_id: [12]
...
           All of the following:
               Gravity Suit
-              Space Jump or Speed Booster
-          All of the following:
-              Hi-Jump and Stand On Frozen Enemies (Intermediate) and Can Freeze Enemies
               Any of the following:
-                  Underwater Wall Jump (Hypermode)
-                  Gravity Suit and Wall Jump (Intermediate)
+                  Space Jump or Speed Booster
+                  # NHJ :https://youtu.be/AWajsYrYgbA
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt:439]: you didnt address this

File: randovania/games/fusion/logic_database/Sector 4 AQA.txt [hunk 14]
```diff
@@ -260,9 +255,9 @@ Extra - room_id: [6]
               # Destroy bomb blocks by standing off of crab: https://youtu.be/h8Q6TKHg8kM
               Stand On Frozen Enemies (Beginner) and Can Freeze Enemies and Can Use Bombs
               Any of the following:
-                  Hi-Jump
-                  # NHJ: https://youtu.be/_F80ASGzrvo
-                  Movement (Advanced) and Wall Jump (Beginner)
+                  Can Use Springball
+                  # NHJ: https://youtu.be/6RhuTs1mD1c
+                  Mid-Air Morph (Beginner) and Movement (Intermediate)
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/games/fusion/logic_database/Sector 4 AQA.txt]: you didnt address this

**Review by Miepee — APPROVED** [review]
